### PR TITLE
feat(over window): generate EOWC stream plan from `LogicalOverAgg`

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR enables [Over Window](https://github.com/risingwavelabs/rfcs/pull/8) feature with `EMIT ON WINDOW CLOSE`. Tracked in #9124.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

### Types of user-facing changes

- SQL commands, functions, and operators

### Release note

Support Over Window (a.k.a Over Aggregation or Window Function) with `EMIT ON WINDOW CLOSE`. Currently only support windows partitioned by at least one column and ordered by only single watermark column.

Example:

```sql
CREATE SOURCE t (a TIMESTAMP, b INT, WATERMARK FOR a AS a - INTERVAL '5 minutes') WITH ...;
CREATE MATERIALIZED VIEW mv EMIT ON WINDOW CLOSE AS
SELECT
    window_start,
    lag(b) OVER (PARTITION BY window_start ORDER BY a),
    max(b) OVER (PARTITION BY window_start ORDER BY a ROWS BETWEEN 10 PRECEDING AND CURRENT ROW)
FROM tumble(t, a, INTERVAL '1 hour');
```

Window functions supported by far:

- `lag` and `lead` with constant offset
- aggregate functions

Window frame types supported by far:

- `ROWS` without exclusion clause

### CODE REVIEW

File: src/stream/src/from_proto/eowc_over_window.rs [hunk 1]
```diff
@@ -56,7 +56,8 @@ impl ExecutorBuilder for EowcOverWindowExecutorBuilder {
                 .expect("vnodes not set for EOWC over window"),
         ));
         let state_table =
-            StateTable::from_table_catalog(node.get_state_table()?, store, vnodes).await;
+            StateTable::from_table_catalog_inconsistent_op(node.get_state_table()?, store, vnodes)
```
Comment (soundOfDestiny) [inline src/stream/src/from_proto/eowc_over_window.rs:59]: yet another inconsistent op

**Review by soundOfDestiny — APPROVED** [review]
lgtm

**Comment by chenzl25** [comment]
> [richardchien](https://github.com/richardchien) wants to merge 6 commits into [rc/eowc-keyword](https://github.com/risingwavelabs/risingwave/tree/rc/eowc-keyword) from [rc/logical-over-window-to-stream](https://github.com/risingwavelabs/risingwave/tree/rc/logical-over-window-to-stream)

Why not merge into the main branch?

**Comment by richardchien** [comment]
> Why not merge into the main branch?

Because this PR depends on #9622.

**Review by chenzl25 — APPROVED** [review]
LGTM

**Pull request merged**